datalad_next/utils/deprecate.py

```diff
@@ -0,0 +1,89 @@
+import inspect
+import warnings
+from functools import wraps
+
+from datalad_next.utils import get_wrapped_class
+
+
+def deprecated(msg, version, parameter=None, parameter_choice=None):
+    """Annotate functions, classes, or function parameters with standardized
+    deprecation warnings.
+
+    Parameters
+    ----------
+    version: str
+      Software version number at which the deprecation was made
+    msg: str
+      Custom message to append to a deprecation warning
+    parameter: str
+      Individual parameter being deprecated (instead of entire function/class)
+    parameter_choice: str
+      Individual parameter choice of specified parameter being deprecated
+    """
+
+    base_template = "{func} was deprecated in version {version}. {msg}"
+    if parameter is None:
```

Comment:
It would make sense to me to move the whole body of this conditional into a dedicated helper function.

I think the two inline `decorator` functions could also become standalone functions. I might give this a try later.

Reply:
Simplified and merged.